pkg/strategy/dca2/recover.go

```diff
@@ -0,0 +1,266 @@
+package dca2
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/core"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type queryAPI interface {
+	QueryOpenOrders(ctx context.Context, symbol string) ([]types.Order, error)
+	QueryClosedOrdersDesc(ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64) ([]types.Order, error)
+	QueryOrderTrades(ctx context.Context, q types.OrderQuery) ([]types.Trade, error)
+}
+
+func (s *Strategy) recover(ctx context.Context) error {
+	s.logger.Info("[DCA] recover")
+	queryService, ok := s.Session.Exchange.(queryAPI)
+	if !ok {
+		return fmt.Errorf("[DCA] exchange %s doesn't support queryAPI interface", s.Session.ExchangeName)
+	}
+
+	openOrders, err := queryService.QueryOpenOrders(ctx, s.Symbol)
+	if err != nil {
+		return err
+	}
+
+	closedOrders, err := queryService.QueryClosedOrdersDesc(ctx, s.Symbol, time.Time{}, time.Now(), 0)
+	if err != nil {
+		return err
+	}
+
+	currentRound, err := getCurrentRoundOrders(s.Short, openOrders, closedOrders, s.OrderGroupID)
+	if err != nil {
+		return err
+	}
+	debugRoundOrders(s.logger, "current", currentRound)
+
+	// recover state
+	state, err := recoverState(ctx, s.Symbol, s.Short, int(s.MaxOrderNum), openOrders, currentRound, s.OrderExecutor.ActiveMakerOrders(), s.OrderExecutor.OrderStore(), s.OrderGroupID)
```

Comment:
do you mean getCurrentRecoverState?

Reply:
Yes, this state means the state where dca2 is at